packages/nx/src/utils/package-manager.ts

```diff
@@ -478,3 +494,120 @@ export async function packageRegistryPack(
   const tarballPath = stdout.trim();
   return { tarballPath };
 }
+
+/**
+ * Gets the workspaces defined in the package manager configuration.
+ * @returns workspaces defined in the package manager configuration, null if not found
+ */
+export function getPackageWorkspaces(
+  packageManager: PackageManager = detectPackageManager(),
+  root: string = workspaceRoot
+): string[] | null {
+  let workspaces: string[] | null = null;
+
+  if (
+    packageManager === 'npm' ||
+    packageManager === 'yarn' ||
+    packageManager === 'bun'
+  ) {
+    const packageJson = readPackageJson(root);
+    workspaces = packageJson.workspaces;
+  } else if (packageManager === 'pnpm') {
+    const pnpmWorkspacePath = join(root, 'pnpm-workspace.yaml');
+    if (existsSync(pnpmWorkspacePath)) {
+      const { packages } =
+        readYamlFile<{ packages: string[] }>(pnpmWorkspacePath) ?? {};
+      workspaces = packages;
+    }
+  }
+
+  return workspaces;
+}
+
+/**
+ * Adds a package to the workspaces defined in the package manager configuration.
+ * If the package is already included in the workspaces, it will not be added again.
+ * @param packageManager The package manager to use. If not provided, it will be detected based on the lock file.
+ * @param root The directory the commands will be ran inside of. Defaults to the current workspace's root.
+ * @param packagePath The path of the package to add to the workspaces
+ * @returns true if the package was added to the workspaces, false otherwise
+ */
+export function addPackagePathToWorkspaces(
+  packageManager: PackageManager = detectPackageManager(),
+  root: string = workspaceRoot,
+  packagePath: string
+): boolean {
+  let workspaces: string[] = getPackageWorkspaces(packageManager, root) ?? [];
+  const isPkgIncluded = workspaces.some((w) => minimatch(packagePath, w));
+  if (isPkgIncluded) {
+    return false;
+  }
```

Comment:
Can we not check this.. and always add it? It's up to the user of this function to check if it already exists.And then this function does not need to return a boolean.